lib/db_adapter.go

```diff
@@ -62,32 +65,132 @@ func (adapter *DbAdapter) GetPostAssociationByAttributes(associationEntry *PostA
 	return DBGetPostAssociationByAttributes(adapter.badgerDb, adapter.snapshot, associationEntry)
 }
 
-func (adapter *DbAdapter) GetUserAssociationsByAttributes(associationQuery *UserAssociationQuery) ([]*UserAssociationEntry, error) {
+func (adapter *DbAdapter) GetUserAssociationsByAttributes(
+	associationQuery *UserAssociationQuery,
+	deletedUtxoAssociationIDs *Set[*BlockHash],
```

Comment:
That way we can return nil from any functions that want to return a set.

Reply:
But will update so don't have to do nil checks.

Reply:
Actually the Set generic only works if stored as a pointer. Otherwise there's weird intermittent behavior where the elements of the set can be re-ordered. Maybe something with copying it. Gonna just pass a pointer for now.